packages/runtime/container-runtime/src/summaryStorageCompressionHooks.ts

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { deflate, inflate } from "pako";
+import { compress, decompress } from "lz4js";
+import { ISummaryContext } from "@fluidframework/driver-definitions";
+import { ISummaryTree, ISnapshotTree, ISummaryBlob, SummaryType } from "@fluidframework/protocol-definitions";
+import { Uint8ArrayToString } from "@fluidframework/common-utils";
+import { getBlobAtPath, listBlobPaths, replaceSummaryObject, SummaryStorageHooks } from "./summaryStorageAdapter";
+import { BlobHeaderBuilder, readBlobHeader, skipHeader, writeBlobHeader } from "./summaryBlobProtocol";
+
+export enum Algorithms {
+    NONE = 1,
+    LZ4 = 2,
+    DEFLATE = 3,
+}
+
+const ALGORITHM_KEY = "ALG";
+
+function cloneSummary(summary: ISummaryTree): ISummaryTree {
+    return JSON.parse(JSON.stringify(summary)) as ISummaryTree;
```

Comment:
Note that JSON.parse(JSON.stringify(..)) won't clone non-JSONable types like Uint8Array.

(I think you may get away with this at the moment, since I think you're replacing all the Uint8Arrays in a second pass.)